rslib/src/scheduler/fsrs/params.rs

```diff
@@ -218,22 +217,24 @@ impl Collection {
 
     pub fn evaluate_params(
         &mut self,
-        params: &Params,
         search: &str,
         ignore_revlogs_before: TimestampMillis,
+        num_of_relearning_steps: usize,
     ) -> Result<ModelEvaluation> {
         let timing = self.timing_today()?;
-        let mut anki_progress = self.new_progress_handler::<ComputeParamsProgress>();
-        let guard = self.search_cards_into_table(search, SortMode::NoOrder)?;
-        let revlogs: Vec<RevlogEntry> = guard
-            .col
-            .storage
-            .get_revlog_entries_for_searched_cards_in_card_order()?;
+        let revlogs = self.revlog_for_srs(search)?;
         let (items, review_count) =
             fsrs_items_for_training(revlogs, timing.next_day_at, ignore_revlogs_before);
+        let mut anki_progress = self.new_progress_handler::<ComputeParamsProgress>();
         anki_progress.state.reviews = review_count as u32;
-        let fsrs = FSRS::new(Some(params))?;
-        Ok(fsrs.evaluate(items, |ip| {
+        let fsrs = FSRS::new(None)?;
+        let input = ComputeParametersInput {
+            train_set: items.clone(),
```

Comment:
I see the `train_set` is `items.clone()`, but `items` since to still be defined by `let (items, review_count) =
            fsrs_items_for_training(revlogs, timing.next_day_at, ignore_revlogs_before);` which didn't change.

It feels like the training items would still be the same items than before ? I don't really see where the `training_set = whole_set \ test_set occurs ?

Reply:
Understood thanks to this comment https://github.com/ankitects/anki/pull/3962#issuecomment-2844712511